tensorboard/plugins/graph/tf_graph_common/node.ts

```diff
@@ -653,6 +653,53 @@ export function getFillForNode(templateIndex, colorBy,
       return isExpanded ?
         colorParams.EXPANDED_COLOR : renderInfo.memoryColor ||
         colorParams.UNKNOWN;
+    case ColorBy.OP_COMPATIBILITY:
+      if (renderInfo.node.type === NodeType.OP) {
+        return ((<OpNode>renderInfo.node).compatible) ?
+            tf.graph.render.OpNodeColors.COMPATIBLE :
+            tf.graph.render.OpNodeColors.INCOMPATIBLE;
+      } else if (renderInfo.node.isGroupNode) {
+        let node = <GroupNode>renderInfo.node;
+
+        let numCompat = node.compatibilityHistogram.compatible;
+        let numIncompat = node.compatibilityHistogram.incompatible
+
+        if (numCompat == 0 && numIncompat == 0) {
+          // Return the hue for unknown device.
+          return colorParams.UNKNOWN;
+        }
+
+        let id = "op-compat-" + node.name;
+        let escapedId = tf.graph.util.escapeQuerySelector(id);
```

Comment:
Any preference for this, or should I keep the original method. I can't take credit though, it is based off the coloring code for ColorBy.DEVICE.